packages/sdk/src/protocols/peer_manager.ts

```diff
@@ -58,7 +61,18 @@ export class PeerManager {
         .map((c) => this.mapConnectionToPeer(c))
     );
 
-    return result[0];
+    const newPeer = result[0];
+
+    if (!newPeer) {
+      log.warn(`requestRenew: Couldn't renew peer ${peerId.toString()}.`);
+      return;
+    }
+
+    log.info(
+      `requestRenew: Renewed peer ${peerId.toString()} to ${newPeer.id.toString()}`
+    );
```

Comment:
can we provide a potential reason of why this could've failed? 
Possibly no peers available to renew with?